UPGRADE.md

```diff
@@ -1,3 +1,51 @@
+# Upgrade to 3.0
+
+## BC BREAK User-provided `PDO` instance is no longer supported
+
+In order to share the same `PDO` instances between DBAL and other components, initialize the connection in DBAL and access it using `Connection::getWrappedConnection()->getWrappedConnection()`.
+
+## BC BREAK: the PDO symbols are no longer part of the DBAL API
+
+1. The support of `PDO::PARAM_*`, `PDO::FETCH_*`, `PDO::CASE_*` and `PDO::PARAM_INPUT_OUTPUT` constants in the DBAL API is removed.
```

Comment:
Can we word this differently? First you could mention that `ParameterType` is the alternative, second when users continnue using `PDO` constants, then they will still work, because we use the same values.

This is something we could a.) document on ParameterType to be true b.) add a test for.

Reply:
> Can we word this differently? First you could mention that ParameterType is the alternative […]

Will do. Should we do it here or we can file a separate ticket? I'm not sure how detailed it should be described in the upgrade notes. A blog post or a migration guide looks like a more proper place for such information.

> This is something we could a.) document on ParameterType to be true b.) add a test for.

I'd be careful here. The values of the DBAL constants are left as the _current_ values of the PDO constants to avoid unnecessary disruption but it doesn't mean they have to be equal. The PDO constants are not supported which is clearly stated in this very sentence.

> second when users continue using `PDO` constants, then they will still work, because we use the same values.

No, this is no longer supported. If users do so, they are on their own. If PHP decides to change the values of PDO or introduces a new constant whose value collides with the value of ours, we won't be able to rule it out.